crates/sui/src/db_tool/db_dump.rs

```diff
@@ -0,0 +1,289 @@
+    /// Open an authority store by directory path
+    pub fn open<P: AsRef<Path>>(
+        primary_path: P,
+        secondary_path: P,
+        db_options: Option<Options>,
+    ) -> Self {
+        let (options, point_lookup) = default_db_options(db_options, None);
+
+        let db = {
+            let path = &primary_path;
+            let s_path = &secondary_path;
+            let db_options = Some(options.clone());
+            let opt_cfs: &[(&str, &rocksdb::Options)] = &[
+                ("objects", &point_lookup),
+                ("transactions", &point_lookup),
+                ("owner_index", &options),
+                ("certificates", &point_lookup),
+                ("pending_execution", &options),
+                ("parent_sync", &options),
+                ("effects", &point_lookup),
+                ("sequenced", &options),
+                ("schedule", &options),
+                ("executed_sequence", &options),
+                ("batches", &options),
+                ("last_consensus_index", &options),
+                ("epochs", &options),
+            ];
+
+            typed_store::rocks::open_cf_opts_secondary(path, Some(s_path), db_options, opt_cfs)
+        }
+        .expect("Cannot open DB.");
+
+        let executed_sequence =
+            DBMap::reopen(&db, Some("executed_sequence")).expect("Cannot open CF.");
+
+        let (
+            objects,
+            owner_index,
+            transactions,
+            certificates,
+            pending_execution,
+            parent_sync,
+            effects,
+            sequenced,
+            schedule,
+            batches,
+            last_consensus_index,
+            epochs,
+        ) = reopen! (
+            &db,
+            "objects";<ObjectKey, Object>,
+            "owner_index";<(Owner, ObjectID), ObjectInfo>,
+            "transactions";<TransactionDigest, TransactionEnvelope<S>>,
+            "certificates";<TransactionDigest, CertifiedTransaction>,
+            "pending_execution";<InternalSequenceNumber, TransactionDigest>,
+            "parent_sync";<ObjectRef, TransactionDigest>,
+            "effects";<TransactionDigest, TransactionEffectsEnvelope<S>>,
+            "sequenced";<(TransactionDigest, ObjectID), SequenceNumber>,
+            "schedule";<ObjectID, SequenceNumber>,
+            "batches";<TxSequenceNumber, SignedBatch>,
+            "last_consensus_index";<u64, ExecutionIndices>,
+            "epochs";<EpochId, EpochInfoLocals>
+        );
+        Self {
+            objects,
+            owner_index,
+            transactions,
+            certificates,
+            pending_execution,
+            parent_sync,
+            effects,
+            sequenced,
+            schedule,
+            executed_sequence,
+            batches,
+            last_consensus_index,
+            epochs,
+        }
+    }
+
+    // TODO: condense with macros
+    pub fn dump(&self, table_name: &str) -> anyhow::Result<BTreeMap<String, String>> {
```

Comment:
Will it be easier to add the dump function in SuiDataStore instead? Will that save us from needing the `SuiDataStoreReadonly`? This copy of SuiDataStore might be quite difficult to maintain.

We can add a feature gate to the dump function if we don't want people using it accidentally.